File: MantenimientoDaxDirectorio/MantenimientoDirectorio/BuscaAlias.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Windows.Forms;
using DattCom;
using Microsoft.VisualBasic.CompilerServices;

namespace directMnt
{
	internal partial class BuscaAlias
	{
		private string Seleccion2 = "";
		private string Seleccion = "";
		private string Selección3 = "";
		private string CodNom = "";
		private string ClioPro;
		private string CodigoRet;
		private Syscod.ManSysnetClass Buscod = new Syscod.ManSysnetClass();
		private string Alias_Renamed;
		private string Codigo;
		private int fila;

		public BuscaAlias()
		{
			InitializeComponent();
		}
		public string Buscando(ref string CodigoEmpresa)
		{
			string BuscandoRet = default;
			Codigo = CodigoEmpresa;
			ShowDialog();
			BuscandoRet = Alias_Renamed;
			return BuscandoRet;
		}

		private void btnbuscar_Click(object sender, EventArgs e)
		{
			Retorna();
		}

		private void btncancelarbusca_Click(object sender, EventArgs e)
		{
			Close();
		}

		private void BuscaAlias_Activated(object sender, EventArgs e)
		{
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 1111


						Input:
								On Error Resume Next

						 */
			if (Operators.CompareString(TxtNombre.Text, "", false) > 0)
			{
				if (ListNombre.RowCount > 2)
				{
				}
				else
				{
					TxtNombre.Focus();
				}
			}
			else
			{
				TxtNombre.Focus();
			}
		}

		private void ListNombre_CellClick(object sender, DataGridViewCellEventArgs e)
		{
			fila = e.RowIndex;
		}

		private void ListNombre_DoubleClick(object sender, EventArgs e)
		{
			Retorna();
		}

		private void ListNombre_KeyDown(object sender, KeyEventArgs e)
		{
			{
				var withBlock = ListNombre;
				if (e.KeyCode == Keys.Return)
				{
					Retorna();
				}
			}
		}

		private void BuscaAlias_Load(object sender, EventArgs e)
		{
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 2191


						Input:
								On Error Resume Next

						 */
			LLenarLista();
		}
		private void LLenarLista()
		{
			string sqlaux;
			sqlaux = "SELECT * From identificacionalias  WHERE codigoempresa ='" + Codigo + "' order by  NombreAlias ";
			var Conectar = new SqlConnection(datosEmpresa.strConxAdcom);
			var datt = new DataTable();
			var dataAd = new SqlDataAdapter(sqlaux, Conectar);
			Conectar.Open();
			dataAd.Fill(datt);
			ListNombre.DataSource = datt;
			ListNombre.ClearSelection();
			Conectar.Close();
			Conectar.Dispose();
			datt.Dispose();
			dataAd.Dispose();
			return;
			hayerrores:
			;

			// ControlaErrores
		}

		private void Retorna()
		{
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 3018


						Input:
								On Error Resume Next

						 */
			CodigoRet = ListNombre.Rows[fila].Cells[0].Value.ToString();
			CodNom = ListNombre.Rows[fila].Cells[1].Value.ToString();
			{
				var withBlock = ListNombre.Rows[fila];
				Alias_Renamed = withBlock.Cells[1].Value.ToString() + "," + withBlock.Cells[2].Value.ToString() + "," + withBlock.Cells[3].Value.ToString() + "," + withBlock.Cells[4].Value.ToString();
			}
			Close();
		}
		private void NombImpresion_Click()
		{
			LLenarLista();
		}

		private void Option_Click(ref short Index)
		{
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 3578


						Input:
								On Error Resume Next

						 */
			switch (Index)
			{
				case var @case when @case == 0:
					{
						Seleccion2 = "";
						break;
					}
				case var case1 when case1 == 1:
					{
						Seleccion2 = " AND tipopersona = 'N' ";
						break;
					}
				case var case2 when case2 == 2:
					{
						Seleccion2 = " AND TIPOPERSONA = 'J' ";
						break;
					}
			}
			LLenarLista();

		}

		private void Option1_Click(ref short Index)
		{
			;
#error Cannot convert OnErrorResumeNextStatementSyntax - see comment for details
			/* Cannot convert OnErrorResumeNextStatementSyntax, CONVERSION ERROR: Conversion for OnErrorResumeNextStatement not implemented, please report this issue in 'On Error Resume Next' at character 3980


						Input:
								On Error Resume Next

						 */
			switch (Index)
			{
				case var @case when @case == 0:
					{
						Seleccion = "";
						break;
					}
				case var case1 when case1 == 1:
					{
						Seleccion = " AND ESCLIENTE <> 0 ";
						break;
					}
				case var case2 when case2 == 2:
					{
						Seleccion = " AND ESPROVEEDOR <> 0 ";
						break;
					}
				case var case3 when case3 == 3:
					{
						Seleccion = " AND ESBANCO <> 0 ";
						break;
					}
				case var case4 when case4 == 4:
					{
						Seleccion = " AND ESEMPLEADO <> 0 ";
						break;
					}
			}
			LLenarLista();
		}

		private void TxtNombre_KeyDown(object sender, KeyEventArgs e)
		{
			short KeyCode = (short)e.KeyCode;
			short Shift = (short)((int)e.KeyData / 0x10000);
			if (KeyCode == (int)Keys.Return)
				LLenarLista();
		}

	}
}

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/BuscaAlias.vb
Imports System.Data.SqlClient
Imports System.Windows.Forms
Imports DattCom
Friend Class BuscaAlias
    Dim Seleccion2 As String = ""
    Dim Seleccion As String = ""
    Dim Selección3 As String = ""
    Dim CodNom As String = ""
    Dim ClioPro As String
    Dim CodigoRet As String
    Dim Buscod As New Syscod.ManSysnetClass
    Dim Alias_Renamed As String
    Dim Codigo As String
    Dim fila As Integer
    Public Function Buscando(ByRef CodigoEmpresa As String) As String
        Codigo = CodigoEmpresa
        Me.ShowDialog()
        Buscando = Alias_Renamed
    End Function

    Private Sub btnbuscar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnbuscar.Click
        Retorna()
    End Sub

    Private Sub btncancelarbusca_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btncancelarbusca.Click
        Me.Close()
    End Sub

    Private Sub BuscaAlias_Activated(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Activated
        On Error Resume Next
        If TxtNombre.Text > "" Then
            If ListNombre.RowCount > 2 Then
            Else
                TxtNombre.Focus()
            End If
        Else
            TxtNombre.Focus()
        End If
    End Sub

    Private Sub ListNombre_CellClick(ByVal sender As Object, ByVal e As System.Windows.Forms.DataGridViewCellEventArgs) Handles ListNombre.CellClick
        fila = e.RowIndex
    End Sub

    Private Sub ListNombre_DoubleClick(ByVal sender As Object, ByVal e As System.EventArgs) Handles ListNombre.DoubleClick
        Retorna()
    End Sub

    Private Sub ListNombre_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles ListNombre.KeyDown
        With ListNombre
            If e.KeyCode = System.Windows.Forms.Keys.Return Then
                Retorna()
            End If
        End With
    End Sub

    Private Sub BuscaAlias_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        On Error Resume Next
        LLenarLista()
    End Sub
    Private Sub LLenarLista()
        Dim sqlaux As String
        sqlaux = "SELECT * From identificacionalias  WHERE codigoempresa ='" & Codigo & "' order by  NombreAlias "
        Dim Conectar As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
        Dim datt As New DataTable
        Dim dataAd As New SqlDataAdapter(sqlaux, Conectar)
        Conectar.Open()
        dataAd.Fill(datt)
        ListNombre.DataSource = datt
        ListNombre.ClearSelection()
        Conectar.Close()
        Conectar.Dispose()
        datt.Dispose()
        dataAd.Dispose()
        Exit Sub
hayerrores:
        'ControlaErrores
    End Sub

    Private Sub Retorna()
        On Error Resume Next
        CodigoRet = ListNombre.Rows(fila).Cells(0).Value.ToString
        CodNom = ListNombre.Rows(fila).Cells(1).Value.ToString
        With ListNombre.Rows(fila)
            Alias_Renamed = .Cells(1).Value.ToString & "," & .Cells(2).Value.ToString & "," & .Cells(3).Value.ToString & "," & .Cells(4).Value.ToString
        End With
        Me.Close()
    End Sub
    Private Sub NombImpresion_Click()
        LLenarLista()
    End Sub

    Private Sub Option_Click(ByRef Index As Short)
        On Error Resume Next
        Select Case Index
            Case Is = 0
                Seleccion2 = ""
            Case Is = 1
                Seleccion2 = " AND tipopersona = 'N' "
            Case Is = 2
                Seleccion2 = " AND TIPOPERSONA = 'J' "
        End Select
        LLenarLista()

    End Sub

    Private Sub Option1_Click(ByRef Index As Short)
        On Error Resume Next
        Select Case Index
            Case Is = 0
                Seleccion = ""
            Case Is = 1
                Seleccion = " AND ESCLIENTE <> 0 "
            Case Is = 2
                Seleccion = " AND ESPROVEEDOR <> 0 "
            Case Is = 3
                Seleccion = " AND ESBANCO <> 0 "
            Case Is = 4
                Seleccion = " AND ESEMPLEADO <> 0 "
        End Select
        LLenarLista()
    End Sub

    Private Sub TxtNombre_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles TxtNombre.KeyDown
        Dim KeyCode As Short = CShort(e.KeyCode)
        Dim Shift As Short = CShort(e.KeyData \ &H10000)
        If KeyCode = System.Windows.Forms.Keys.Return Then LLenarLista()
    End Sub

End Class